tests/pyfunc/docker/conftest.py

```diff
@@ -0,0 +1,56 @@
+import os
+from functools import lru_cache
+
+import pytest
+import requests
+from packaging.version import Version
+
+import docker
+import mlflow
+
+TEST_IMAGE_NAME = "test_image"
+MLFLOW_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))
+RESOURCE_DIR = os.path.join(MLFLOW_ROOT, "tests", "resources", "dockerfile")
+
+docker_client = docker.from_env()
+
+
+@pytest.fixture(autouse=True)
+def clean_up_docker_image():
+    try:
+        # Get all containers using the test image
+        containers = docker_client.containers.list(filters={"ancestor": TEST_IMAGE_NAME})
+        for container in containers:
+            container.remove(force=True)
+
+        # Clean up the image
+        docker_client.images.remove(TEST_IMAGE_NAME, force=True)
+    except Exception:
+        pass
```

Comment:
Do we need this try-catch?

Reply:
Yes, If test failed before creating docker image, this clean up throws exception and hides what actually failed in the test.

Reply:
Is it possible to check if the image exists?

Reply:
We can't directly check with image_name (`docker_client.image.get(xxx)` does similar but raises an exception if not exists). We can use `docker_client.image.list()` and search it, but imo it's simpler to just remove inside try-catch as long as it does no harm.

Reply:
Can we catch `docker.errors.ImageNotFound` instead of `Exception`?

Reply:
`docker_client.containers.list` seems to return an empty list when `filters` matches nothing. Can we pull this out from try-catch?

Reply:
yeah scoping down try-catch makes sense!